packages/validator/src/convertor.ts

```diff
@@ -0,0 +1,98 @@
+import { Log, Result } from './types';
+import * as logger from './utils/logger';
+import fs from 'fs';
+import { processSpecs } from './processor';
+
+const oas2ois = 'templates/OAS2OIS.json';
+const ois2cs = 'templates/OIS2C&S.json';
+
+function invalidConversionMessage(from, to): Log {
+  return logger.error(`Conversion from ${from} to ${to} is not valid conversion`);
+}
+
+/**
+ * Converts specification from provided file into format the template specifies
+ * @param specsPath - specification file to convert, root must be an object (not an array)
+ * @param templatePath - template json file
+ * @returns array of messages and converted specification
+ */
+export function convert(specsPath: string | undefined, templatePath: string | undefined): Result {
+  if (!specsPath || !templatePath) {
+    return { valid: false, messages: [logger.error('Specification and template file must be provided')], output: {} };
+  }
+
+  let template, specs;
+
+  try {
+    template = fs.readFileSync(templatePath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${templatePath}`)], output: {} };
```

Comment:
nit: is there a reason you don't just propagate the error message `e`?

Reply:
I dont really know how will be validator used/invoked, I was focusing on scenario in which a nice web UI printing all standard validator messages is made, user selects corrupted file, validator will return message in standard format, UI will display this as any other message, instead of printing 'Something went wrong' or some stacktrace as it didnt receive response in standard format, also no additional work will be needed on frontend.